source/agora/consensus/protocol/Nominator.d

```diff
@@ -761,26 +761,23 @@ private struct SCPEnvelopeHash
     // empty envelope
     import std.conv;
     assert(getEnvHash().hashFull() == Hash.fromString(
-        "0xd3dc2318365e55ea3a62b403fcbe22d447402741a5151a703326dbf852350dcd0" ~
-        "e020a762ae12a871473aed80d82f51c1cd3942c0b2360c2d609279a2867fe68"),
+        "0xfd2ea2b85d2a315a9817e6661bc3c4378637de37649d2fdb9ca82d6e4172e9e46" ~
```

Comment:
The reason why this changed is because the signature can no longer be empty when it's a fixed-length array. It wasn't obvious at first.

Reply:
I think the length of 'xdr:::opaque_vec' is recorded in front  when it is serialized.

Reply:
It's not about serialization, just about hashing. In this specific case the unittest was testing what happens when the signature is empty, because a dynamic array can be empty. But now that the signature is a fixed length array, it's not possible to test for an "empty" signature.

Anyway your code is correct, I just made a comment in case anyone else is confused why the hash changed.

Reply:
Yeah, I checked. Thank you.